gui/src/app/Stanc/Linting.ts

```diff
@@ -0,0 +1,117 @@
+import { StancErrors } from "./Types";
+
+type Position = {
+  startLineNumber: number;
+  startColumn: number;
+  endLineNumber: number;
+  endColumn: number;
+};
+
+// An interface for passing markers (squiggles) to the editor without depending on monaco types
+export type CodeMarker = Position & {
+  message: string;
+  severity: "error" | "warning" | "hint" | "info";
+};
+
+export const stancErrorsToCodeMarkers = (stancErrors: StancErrors) => {
+  const codeMarkers: CodeMarker[] = [];
+
+  for (const x of stancErrors.errors || []) {
+    const marker = stancMessageToMarker(x, "error");
+    if (marker) codeMarkers.push(marker);
+  }
+  for (const x of stancErrors.warnings || []) {
+    const marker = stancMessageToMarker(x, "warning");
+    if (marker) codeMarkers.push(marker);
+  }
+
+  return codeMarkers;
+};
+
+const stancMessageToMarker = (
+  message: string,
+  severity: "error" | "warning",
+): CodeMarker | undefined => {
+  const position = locationFromMessage(message);
+  if (position === undefined) return undefined;
+  const { startLineNumber, startColumn, endLineNumber, endColumn } = position;
+
+  return {
+    startLineNumber,
+    startColumn,
+    endLineNumber,
+    endColumn,
+    message:
+      severity === "warning"
+        ? getWarningMessage(message)
+        : getErrorMessage(message),
+    severity,
+  };
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Adapted from https://github.com/WardBrian/vscode-stan-extension
+
+const locationFromMessage = (message: string): Position | undefined => {
+  if (!message) return;
+  // format is "in 'filename', line (#)), column (#) to (line #,)? column (#)"
+  const start = message.matchAll(/'.*', line (\d+), column (\d+)( to)?/g);
+  if (!start) {
+    return undefined;
+  }
+  // there will be multiple in the case of #included files
+  const lastMatch = Array.from(start).pop();
+  if (!lastMatch) {
+    return undefined;
+  }
+
+  const startLineNumber = parseInt(lastMatch[1]);
+  const startColumn = parseInt(lastMatch[2]) + 1;
+
+  let endLineNumber = startLineNumber;
+  let endColumn = startColumn;
+
+  if (lastMatch[3]) {
+    // " to" was matched
+    const end = message.match(/to (line (\d+), )?column (\d+)/);
+    if (end) {
+      if (end[1]) {
+        endLineNumber = parseInt(end[2]);
+      }
+      endColumn = parseInt(end[3]) + 1;
+    }
+  }
+
+  return { startLineNumber, startColumn, endLineNumber, endColumn };
+};
+
+function getWarningMessage(message: string) {
```

Comment:
Both of these were a result of this code coming from https://github.com/WardBrian/vscode-stan-extension, but since we're already making changes to it we might as well make this one I suppose